core/runtime/common/const_wasm_provider.cpp

```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/common/const_wasm_provider.hpp"
+
+namespace kagome::runtime {
+
+  ConstWasmProvider::ConstWasmProvider(common::Buffer code)
```

Comment:
How about to force using move-semantics for avoid redundant copy?
```suggestion
  ConstWasmProvider::ConstWasmProvider(common::Buffer&& code)
```

Reply:
Once again, will there be one? As far as I know the only difference is that with an rvalue-reference I won't be able to copy a buffer with code without explicitly creating a copy object, and with passing by value it will make a copy if passed an lvalue, but will move rvalues without copying. Please correct me if I am wrong, I will have to seriously reconsider my knowledge of move semantics in this case.

Reply:
With rvalue-arg call with lvalue will be forbidded. Well be allowed call with temporary object (created emplace) or moved value (with using `std::move(blabla)`).

Reply:
Sure, but honestly I'm not sure if I should prohibit calling it with lvalue, because it's just an optimization, not a part of the class contract, I'd prefer to leave it to the caller.